torch/csrc/jit/script/module.h

```diff
@@ -125,17 +126,117 @@ struct TORCH_API Method {
   }
 
  private:
-  // Methods are uniqued onwed by a single module. This raw pointer allows
-  // looking up the module.
-  ModulePtr owner_;
+  // Methods are uniqued onwed by a single object. This raw pointer allows
+  // looking up the object.
+  ObjectPtr owner_;
 
   // Underlying unbound function
   // This is the _lowered_ function and is different than the
   // first-class function in class_compilation_unit()
   Function* function_;
 };
 
-struct TORCH_API Module {
+struct TORCH_API Object {
+  Object() {}
+  Object(ObjectPtr object_value) : object_value_(std::move(object_value)) {}
+
+  const c10::QualifiedName& name() const {
+    return *object_value()->type()->name();
+  }
+
+  void setattr(const std::string& name, IValue v) {
+    size_t slot = object_value()->type()->getAttributeSlot(name);
+    const TypePtr& expected = object_value()->type()->getAttribute(slot);
+    TORCH_CHECK(
+        v.type()->isSubtypeOf(expected),
+        "Expected a value of type '",
+        expected->python_str(),
+        "' for field '",
+        name,
+        "', but found '",
+        v.type()->python_str(),
+        "'");
+    object_value()->setSlot(slot, std::move(v));
+  }
+
+  ObjectPtr object_value() const;
```

Comment:
Since we are going to go through the effort of renaming `module_value`, I think this should be called `_ivalue`. In the future, we are not going to expose this publically.